src/NexusMods.App.UI/WorkspaceSystem/Page/PageFactoryController.cs

```diff
@@ -44,8 +45,29 @@ public IPageFactory GetFactory(PageData pageData)
     /// <exception cref="KeyNotFoundException">Thrown when there is no registered factory with ID <see cref="PageData.FactoryId"/></exception>
     public Page Create(PageData pageData, WindowId windowId, WorkspaceId workspaceId, PanelId panelId, Optional<PanelTabId> tabId)
     {
-        var factory = GetFactory(pageData);
-        var page = factory.Create(pageData.Context);
+        Page page;
+
+        try
+        {
+            var factory = GetFactory(pageData);
+            page = factory.Create(pageData.Context);
+        }
+        catch (Exception e)
+        {
+            var errorPageData = new PageData
+            {
+                FactoryId = ErrorPageFactory.StaticId,
+                Context = new ErrorPageContext
+                {
+                    Exception = e,
+                    FailedPage = pageData,
+                },
+            };
+
+            var factory = GetFactory(errorPageData);
+            page = factory.Create(errorPageData.Context);
+        }
```

Comment:
Wouldn't it have been better to handle the exception in the callers and just log an error so that an exception dialog is surfaced?